### DISCUSSION TITLE
当 load balancing 切换服务器的时候，当前活动连接是否会中断？

### DISCUSSION THREAD

**Discussion by pexcn** [shadowsocks/shadowsocks-rust#1111] [Q&A] [answered]
当服务器出现如下日志的时候：
```sh
2023-02-10T23:26:09.334097264+08:00 INFO  switched best TCP server from xx.xx.xx.xx:1984 to yy.yy.yy.yy:1984
2023-02-10T23:26:29.338321573+08:00 INFO  switched best UDP server from xx.xx.xx.xx:1984 to yy.yy.yy.yy:1984
```

对这个功能有两个疑问：
1. 如果我正在进行在线联机游戏 (UDP)，是否会导致退出游戏房间呢？
2. 如果我在服务器切换之前，（局域网透明代理，ss-rust 的 `timeout` 设置为 `600`）客户端使用了 `curl ip-api.com` 会显示 xx.xx.xx.xx 的 IP, 而服务器切换之后，在 timeout 超时之前，我再执行 `curl ip-api.com`, 是否会显示切换之后的 IP 呢？

cc @zonyitoo 😅

**Comment by zonyitoo** [comment]
It won't change any already established connections.

**Comment by database64128** [comment, marked as answer]
When the load balancer switches servers, it does not affect established TCP connections and UDP associations. Now, about your questions:

1. It depends on how the game uses UDP sockets. If the game does not create and switch to new UDP sockets after the server switch, then it won't be affected at all. If the game does create new UDP sockets, but does not care about source IP changes, then you'll be fine too.
2. Each time you run `curl`, it opens a new socket. Previous runs of `curl` have no lasting effect on subsequent runs.

**Comment by pexcn** [comment]
感谢大佬们的解答 😁
